contrib/systemd/mako.service.in

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Lightweight Wayland notification daemon
+Documentation=man:mako(1)
+PartOf=graphical-session.target
```

Comment:
Ho I forgot, I think we also want a `Requires` and an `After` dependencies. I got nasty race conditions with `sway-services` where the services were started before the environment was fully populated.

Reply:
yes. I haven't updated my profile either. After is needed.

Reply:
Hmm... the service will only be started once the first notification is sent to it.

Maybe the problem is that you're trying to trigger notifications before the environment was fully populated?

Reply:
If the service is enabled, It will be started right away

Reply:
I didn't check this with upstream systemd, but the observed behaviour suggests that systemd makes the dependancy tree first (declaratively, no need to run anything), and if mako doesn't specify ordering it'll just start it in parallel to everything.

Actually, now that I think about it, that's how systemd says it works.

ps. yes I enable the mako service, I don't want any other notification daemon dbus activated by chance.

Reply:
This service is auto-started on-demand by DBus-activation.

You should not `enable` it. That will just make your startup a bit slower. I wish that `RefuseManualStart` would be compatible with DBus-activation, since I've many times seen people eager-start dbus services. :disappointed: 

Even so, following the setup instructions, in, eg, sway's wiki should not lead to race conditions. `sway` would run before this target is considered reached (after `graphical-session-pre`, actually).

If you have a very custom setup with many notification-daemons installed, but want to prioritise that, I do think you should have to just manually override things locally. I'd generally prefer to give the best experience out-of-the-box to standard setups, and just allow very odd or highly customised setups to override locally as they need.
Depending on a custom non-upstream target is the opposite: it works for your specific custom setup, but won't work for people out-of-the-box on standard setups, and will lead to the need to document extra steps _only_ if you're using the systemd standard target designed specifically for this.

Reply:
> You should not enable it.

ps. yes I enable the mako service, I don't want any other notification daemon dbus activated by chance. :)